### ISSUE TITLE
Add Metadata to Alerts for Traceability and Integration

### ISSUE THREAD

**Issue by mastercactapus** [target/goalert#3653]
**What problem would you like to solve? Please describe:**
Difficulties in tracing and integrating with other systems due to the lack of ability to associate arbitrary metadata with an alert upon its creation and retrieval.

**Describe the solution you'd like:**
- Implement a new `meta` in alert creation endpoints to accommodate a set of `key=value` pairs. This should also apply to the `createAlert` mutation input.
- Create a new `alert_data` table for storing the new data, with an `alert_id` column and a data column (`jsonb`).
- Update the UI to display these key-value pairs.
- Extend Webhook notifications to include the new `meta` values.

Example of the Graphql input and struct formats:

```graphql
input AlertMetadataInput {
  key: String!
  value: String!
}

input CreateAlertInput {
  // ...
  meta: [AlertMetadataInput!]
}

type AlertMetadata {
  key: String!
  value: String!
}

type Alert {
  // ...
  meta: [AlertMetadata!]
}
```

```go
type AlertMetaValue struct {
	Key string `json:"key"`
	Value string `json:"value"`
}

const TypeAlertMetaV1 = "alert_meta_v1"

type AlertMeta struct {
	Type string `json:"type"`
	AlertMetaV1 []AlertMetaValue `alert_meta_v1`
}
```

**Additional context:**
This feature can enhance the handling of alerts to make them compatible with query parameters as well as GraphQL.

**Comment by Forfold** [comment]
@mastercactapus What do you envision `Type string `json:"type"`` being within `AlertMeta`?

**Comment by mastercactapus** [comment]
@Forfold That struct is specifically what's going to Postgres -- the idea is to have type/version info embedded in it so we have something to key off of in any migrations/changes we need to make down the line.

It wouldn't be included in any of the API/external stuff, just internal

**Comment by Forfold** [comment]
@mastercactapus Gotcha, so the external integration wouldn't ever see this value, as it would be a hardcoded string set in the function at time of insert/update?

**Comment by mastercactapus** [comment]
@Forfold yep, exactly. Not only that, but the `alert.Store` method shouldn't even expose it, it's just to ensure integrity within the store method to the DB (since things change over time).

**Comment by mastercactapus** [comment]
Also, a nice addition would be adding a `metaValue(key: string): string` field to `Alert` in the GraphQL schema. It would play nice with GraphQL, allowing mapping (potentially multiple) metadata fields in the return structure, for example:
```graphql
alert(id: 123) {
  myData: metaValue(key: "my_data_v1")
}
```

would return:
```json
{"data": {
  "alert": {
    "id": 123,
    "myData": "some value"
  }
}}
```

(thanks @Forfold for helping with this idea!)

**Comment by mastercactapus** [comment]
For data storage, it may be better to store as a map so that we leave open the possibility to index these (for search) in the future. Part of validation would need to ensure unique keys anyway:

```go
const TypeAlertMetaV1 = "alert_meta_v1"

type AlertMeta struct {
	Type string `json:"type"`
	AlertMetaV1 map[string]string `alert_meta_v1`
}
```

**Comment by jinjaghost** [comment]
Hi,

To better track an alert in the UI, it would be interesting to show the metadata (if present / in a dedicated section ?) in the Alert Page. How would you implement this ? If you provide hints i can try to make a PR.

Thanks

**Comment by mastercactapus** [comment]
Left TODO:
- [ ] Metadata support for non-JSON generic API calls
- [ ] Metadata support for UIK
  - add `meta` as a param -- should produce a JSON string that is map[string]string
- [ ] UI should display metadata for alerts
